File: src/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */

export function maxArea(height) {
  let max = 0,
    left = 0,
    right = height.length - 1
  while (left < right) {
    let tmpMax = Math.min(height[left], height[right]) * (right - left)
    max = Math.max(max, tmpMax)
    if (height[left] > height[right]) right--
    else left++
  }
  return max
}

File: src/11. Container With Most Water.cpp
class Solution {
    public:
        int maxArea(vector < int > & height) {
            int maxAmount = 0;
            int left = 0;
            int right = height.size() - 1;
            while (left < right) {
                int localMaxAmount = (right - left) * min(height[left], height[right]);
                maxAmount = max(maxAmount, localMaxAmount);
                if (height[left] > height[right]) {
                    right--;
                } else {
                    left++;
                }
            }
            return maxAmount;
        }
};